src/protocol/amcp/amcp_command_repository.cpp

```diff
@@ -76,18 +76,21 @@ struct amcp_command_repository::impl
     std::map<std::wstring, std::pair<amcp_command_func, int>> commands;
     std::map<std::wstring, std::pair<amcp_command_func, int>> channel_commands;
 
-    impl(const std::vector<spl::shared_ptr<core::video_channel>>&    channels,
-         const spl::shared_ptr<core::cg_producer_registry>&          cg_registry,
+    impl(const spl::shared_ptr<core::cg_producer_registry>&          cg_registry,
          const spl::shared_ptr<const core::frame_producer_registry>& producer_registry,
          const spl::shared_ptr<const core::frame_consumer_registry>& consumer_registry,
          std::function<void(bool)>                                   shutdown_server_now)
         : cg_registry(cg_registry)
         , producer_registry(producer_registry)
         , consumer_registry(consumer_registry)
         , shutdown_server_now(shutdown_server_now)
+    {
+    }
+
+    void init(const std::vector<spl::shared_ptr<core::video_channel>>& channs)
     {
         int index = 0;
-        for (const auto& channel : channels) {
+        for (const auto& channel : channs) {
```

Comment:
I'm not a fan of short names, why `channs` instead of `channels`?

Reply:
Because of a `warning treated as error: declaration of 'channels' hides class member`. Would calling the param 'video_channels' be better? I haven't paid enough attention to naming - I'll work on code quality in this PR.